beluga/test/beluga/sensor/test_likelihood_3d_field_model.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright 2024 Ekumen, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <utility>
+#include <vector>
+
+#include <openvdb/openvdb.h>
+#include <openvdb/tools/Diagnostics.h>
+#include <openvdb/tools/TopologyToLevelSet.h>
+
+#include <range/v3/range/conversion.hpp>
+#include <range/v3/view/transform.hpp>
+#include <sophus/common.hpp>
+
+#include "beluga/sensor/likelihood_3d_field_model.hpp"
+#include "beluga/test/simple_pointcloud_interface.hpp"
+
+namespace {
+
+template <typename GridT, typename T>
+auto make_map(const double voxel_size, const std::vector<T>& world_points) {
+  // Create the grid
+  typename GridT::Ptr grid = openvdb::FloatGrid::create();
+  grid->setTransform(openvdb::math::Transform::createLinearTransform(voxel_size));
+
+  // Get an accessor for coordinate-based access to voxels
+  typename GridT::Accessor accessor = grid->getAccessor();
+
+  // Fill the grid
+  const openvdb::math::Transform& transform = grid->transform();
+  for (const auto& point : world_points) {
+    // Transform to index world
+    const openvdb::math::Coord ijk = transform.worldToIndexCellCentered(point);
+    // Set the voxel to 1
+    accessor.setValue(ijk, 1.0);
+    // Activate the voxel
+    accessor.setActiveState(ijk);
+  }
+
+  // Check grid
+  openvdb::tools::CheckLevelSet<GridT> checker(*grid);
+  // Check inactive values have a magnitude equal to the background value
+  assert(checker.checkInactiveValues() == "");
+
+  // Associate metadata
+  grid->setName("map");
+
+  // Transform to levelset
+  typename GridT::Ptr grid_levelset = openvdb::tools::topologyToLevelSet(*grid, 1, 0, 0, 0);
+
+  return grid_levelset;
```

Comment:
@pvela2017 consider collapsing these lines into one:
```suggestion
  return openvdb::tools::topologyToLevelSet(*grid, 1, 0, 0, 0);
```

Reply:
Changed in ee56b7f6b2bfe9f580931acb8ecc0e617a2d91c9